src/plugins/intel_gna/ops/util/util.hpp

```diff
@@ -139,6 +157,148 @@ inline bool is_one_dim_shapes(const ov::Shape& in_dims, const ov::Shape& out_dim
     return is_one_dim_shape(in_dims) && is_one_dim_shape(out_dims);
 }
 
+static bool is_power_activation(const ov::Node* node) noexcept {
+    if (auto power_op = dynamic_cast<const ngraph::opset9::Power*>(node)) {
+        auto const_node = std::dynamic_pointer_cast<ngraph::opset9::Constant>(power_op->get_input_node_shared_ptr(1));
+        if (!const_node)
+            return false;
+        float value;
+        if (!ngraph::op::util::get_single_value(const_node, value)) {
+            return true;
+        }
+        return (1.0f != value);
+    } else if (auto power_op = std::dynamic_pointer_cast<ngraph::op::PowerIE>(node)) {
+        return (1.0f != power_op->power);
+    }
+    return false;
+}
+
+static bool is_power_activation(const std::shared_ptr<ngraph::Node>& node) noexcept {
+    return is_power_activation(node.get());
+}
+
+static bool is_eltwise_mul(const ngraph::Output<ngraph::Node>& node) {
+    auto eltwise = std::dynamic_pointer_cast<ngraph::op::Eltwise>(node.get_node_shared_ptr());
+    if (!eltwise) return false;
+    return eltwise->eltwise_type == ELTWISE_TYPE::Prod;
+}
+
+static bool is_eltwise_add(const ngraph::Output<ngraph::Node>& node) {
+    auto eltwise = std::dynamic_pointer_cast<ngraph::op::Eltwise>(node.get_node_shared_ptr());
+    if (!eltwise) return false;
+    return eltwise->eltwise_type == ELTWISE_TYPE::Sum;
+}
+
+static bool is_pooling(const ngraph::Output<ngraph::Node>& node) {
+    return (std::dynamic_pointer_cast<ngraph::opset7::MaxPool>(node.get_node_shared_ptr()) != nullptr);
+}
+
+static bool is_applied(const std::vector<std::function<bool()>> functions) {
+    for (auto&& function : functions) {
+        if (function()) {
+            return true;
+        }
+    }
+    return false;
+}
+
+template <typename T>
+static bool is_Tbit_fq(const std::shared_ptr<ngraph::Node>& node) {
+    auto fq_node = std::dynamic_pointer_cast<ngraph::opset9::FakeQuantize>(node);
+    if (!fq_node)
+        return false;
+    auto levels = fq_node->get_levels();
+    return std::numeric_limits<T>::max() == levels;
+}
+
+static bool is_32bit_fq(const std::shared_ptr<ngraph::Node>& node) {
+    return is_Tbit_fq<uint32_t>(node);
+}
+
+static bool is_16bit_fq(const std::shared_ptr<ngraph::Node>& node) {
+    return is_Tbit_fq<uint16_t>(node);
+}
+
+static bool is_8bit_fq(const std::shared_ptr<ngraph::Node>& node) {
+    return is_Tbit_fq<uint8_t>(node);
+}
+
+static bool is_activation(const ov::Node* node) noexcept {
+    const std::vector<std::function<bool()>> activations = {
+            [node]() { return dynamic_cast<const ngraph::opset9::Clamp*>(node) != nullptr; },
```

Comment:
just wonder if this is common approach for checking if operation is of specific type?
This will probably lead to creation of vector of lambdas which will be executed later in for look.
Wouldn't be better to make it static or eventually replace it with if/elseif ... ?